plugin/backend.go

```diff
@@ -18,7 +19,7 @@ type ServiceBackend interface {
 	Reverse(state request.Request, exact bool, opt Options) ([]msg.Service, error)
 
 	// Lookup is used to find records else where.
-	Lookup(state request.Request, name string, typ uint16) (*dns.Msg, error)
+	Lookup(state request.Request, name string, typ uint16, opt Options) (*dns.Msg, error)
```

Comment:
Should we fumble context.Context in state? `opt` makes sense as well, but context.Context is not really an option.
opt is also nice because of the symmetry with `Reverse`

Reply:
fumble? What are you trying to suggest?

Reply:
should request.Request get a context.Context member in the struct.

Reply:
Oh, you mean "bundle" :)

Yeah, that could work. `opt` already existed in some places which is why Chris used it, I don't know if that place uses request.Request or not - @chrisohaver ?

Reply:
Yeah, in all these https://github.com/coredns/coredns/pull/1484/files#diff-347a95a2d36b7b336cb55cc146bd16d6R36 - they have `state` too, so that would work and be less change. But what's the point of Options in all those...

Reply:
yeah - its not really an "option."  I'll add it to request.Request.